proposals/0000-unicode-for-string-processing.md

```diff
@@ -0,0 +1,1321 @@
+
+```swift
+let regex4 = Regex {
+    "ba"
+    "na".ignoresCase(false)
+    "na"
+}
+.ignoresCase()
+
+"banana".contains(regex4)     // true
+"BAnaNA".contains(regex4)     // true
+"BANANA".contains(regex4)     // false
+
+// Equivalent to:
+let regex5 = /(?i)ba(?-i:na)na/
+```
+
+All option APIs are provided on `RegexComponent`, so they can be called on a `Regex` instance, or on any component that you would use inside a `RegexBuilder` block when the `RegexBuilder` module is imported.
+
+The options that `Regex` supports are shown in the table below, in three groups: Options that affect matching behavior for _both regex syntax and APIs_, options that affect the matching behavior of _regex syntax only_, and options with _structural_ or _syntactic_ effects that are only supported through regex syntax.
+
+| **Matching Behavior**        |                |                                 | Default            |
+|------------------------------|----------------|---------------------------------|--------------------|
+| Case insensitivity           | `(?i)`         | `ignoresCase()`                 | disabled           |
+| ASCII-only character classes | `(?DSWP)`      | `asciiOnlyClasses(_:)`          | `.none`            |
+| Unicode word boundaries      | `(?w)`         | `wordBoundaryKind(_:)`          | `.default`         |
+| Semantic level               | `(?Xu)`        | `matchingSemantics(_:)`         | `.graphemeCluster` |
+| Default repetition behavior  | n/a            | `defaultRepetitionBehavior(_:)` | `.eager`           |
+| **Regex Syntax Only**        |                |                                 |                    |
+| Single-line mode             | `(?s)`         | `dotMatchesNewlines()`          | disabled           |
+| Multi-line mode              | `(?m)`         | `anchorsMatchNewlines()`        | disabled           |
+| Swap eager/reluctant         | `(?U)`         | n/a                             | disabled           |
+| **Structural/Syntactic**     |                |                                 |                    |
+| Extended syntax              | `(?x)`,`(?xx)` | n/a                             | `xx` enabled in multi-line regex literals; otherwise, off |
+| Named captures only          | `(?n)`         | n/a                             | disabled           |
+
+#### Case insensitivity
+
+Regexes perform case sensitive comparisons by default. The `i` option or the `ignoresCase(_:)` method enables case insensitive comparison.
+
+```swift
+let str = "Café"
+	
+str.firstMatch(of: /CAFÉ/)          // nil
+str.firstMatch(of: /(?i)CAFÉ/)      // "Café"
+str.firstMatch(of: /(?i)cAfÉ/)      // "Café"
+```
```

Comment:
This seems too interior-syntax heavy and we don't get to see the API name in use. What about having an additional line that uses `.ignoresCase()` after the literal?

Reply:
Sure, I'll update to include API usage throughout this section. The goal is here is dual — we're explaining the semantics of the option and the two ways of applying it.